src/Redux_Todos_Thunk/components/TodoList.js

```diff
@@ -0,0 +1,82 @@
+import React, { useState } from "react";
+import { connect } from "react-redux";
+import TodoItem from "./TodoItem";
+import {
+  FILTER_ALL,
+  FILTER_ACTIVE,
+  FILTER_COMPLETED,
+} from "../constants/filterTypes";
+
+const TodoList = ({ todos }) => {
+  const [filter, setFilter] = useState(FILTER_ALL);
+  const [currentPage, setCurrentPage] = useState(1);
+  const todosPerPage = 5;
+
+  const handlePageChange = (pageNumber) => {
+    setCurrentPage(pageNumber);
+  };
+
+  const handleFilterChange = (newFilter) => {
+    setFilter(newFilter);
+    setCurrentPage(1);
+  };
+
+  const filteredTodos = () => {
+    switch (filter) {
+      case FILTER_ACTIVE:
+        return todos.filter((todo) => !todo.completed);
+      case FILTER_COMPLETED:
+        return todos.filter((todo) => todo.completed);
+      default:
+        return todos;
+    }
+  };
+
+  const indexOfLastTodo = currentPage * todosPerPage;
+  const indexOfFirstTodo = indexOfLastTodo - todosPerPage;
+  const currentTodos = filteredTodos().slice(indexOfFirstTodo, indexOfLastTodo);
+
+  const totalPages = Math.ceil(filteredTodos().length / todosPerPage);
+
+  return (
+    <div>
+      <div>
```

Comment:
Make separate component `Pagination` without internal state

The props: `total`, `page`, `limit` (`totalTodosCount`, `currentPage`, `todosPerPage`)

Following data `totalTodosCount`, `currentPage`, and `todosPerPage` should live in the reducer, calculated and passed through `connect()`

Reply:
Also, add to pagination additional buttons for switch limit per page

Allow for user to switch between `5, 10, 15` and see selected limit. of todos on a page

You need additional handler

```
onChangeLimit: (limit: number) => void
```

Reply:
In real world

Frontend sends to backend `page` and `limit` when request lists

Backend sends to frontend `items` and `total` when response on requested lists